tasks/omnibus.py

```diff
@@ -265,9 +266,13 @@ def build(
                 with timed(quiet=True) as restore_cache:
                     # Allow failure in case the cache was evicted
                     if ctx.run(f"{aws_cmd} s3 cp --only-show-errors {git_cache_url} {bundle_path}", warn=True):
-                        print(f'Successfully restored cache {cache_key}')
-                        ctx.run(f"git clone --mirror {bundle_path} {omnibus_cache_dir}", err_stream=sys.stdout)
-                        cache_state = ctx.run(f"git -C {omnibus_cache_dir} tag -l").stdout
+                        print(f'Successfully retrieved cache {cache_key}')
+                        try:
+                            ctx.run(f"git clone --mirror {bundle_path} {omnibus_cache_dir}")
+                        except UnexpectedExit as exc:
+                            print(f"An error occurring while cloning the cache repo: {exc}")
+                        else:
+                            cache_state = ctx.run(f"git -C {omnibus_cache_dir} tag -l").stdout
                     else:
                         print(f'Failed to restore cache from key {cache_key}')
                         send_cache_miss_event(
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏technically we could end up here as a cache hit but hitting the corruption so these events would be somewhat inaccurate. We'll maybe want to come back to that to improve visibility if we continue having problems.

Reply:
Would we? We only arrive here if downloading fails; the corruption errors on the `git clone`, thus we wouldn't be reporting corruptions as cache misses. Though I agree on the visibility if we still find it problematic. Or maybe I misunderstood what you mean?

Reply:
I think I just misunderstood the control flow here, I thought the `except` dropped us into the outer `else`. I think @Pythyu's comment about another metric/event could be useful, but doesn't have to be a part of this PR, imho.